server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -662,6 +662,20 @@ float score(float similarity, ElementType elementType, int dim) {
                     case FLOAT -> (1 + similarity) / 2f;
                 };
             }
+        },
+        MAX_INNER_PRODUCT(VectorSimilarityFunction.MAXIMUM_INNER_PRODUCT) {
+            @Override
+            float score(float similarity, ElementType elementType, int dim) {
+                return switch (elementType) {
+                    case BYTE, FLOAT -> {
+                        if (similarity < 1) {
+                            yield -1 * (1f / similarity - 1);
+                        } else {
+                            yield similarity - 1f;
```

Comment:
should this be `similarity + 1f`?

Reply:
no, this is transforming the score back to the actual maximum inner product. So it is the inverse of the mip transformation in Lucene:
```
  public static float scaleMaxInnerProductScore(float vectorDotProductSimilarity) {
    if (vectorDotProductSimilarity < 0) {
      return 1 / (1 + -1 * vectorDotProductSimilarity);
    }
    return vectorDotProductSimilarity + 1;
  }
```

Reply:
@benwtrent This is not how I interpreted this `score` function; my interpretation is that given a similarity it produces a score based from similarity. This is also in line how it is implemented for other similarity?

Are things different for maxInnerProduct?